langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchInformationResult.java

```diff
@@ -0,0 +1,140 @@
+     *
+     * @param totalResults The total number of results.
+     * @param pageNumber  The page number.
+     * @param metadata     The metadata.
+     */
+    public WebSearchInformationResult(Long totalResults, Integer pageNumber, Map<String, Object> metadata) {
+        this.totalResults = ensureNotNull(totalResults, "totalResults");
+        this.pageNumber = pageNumber;
+        this.metadata = metadata;
+    }
+
+    /**
+     * Gets the total number of results.
+     *
+     * @return The total number of results.
+     */
+    public Long totalResults() {
+        return totalResults;
+    }
+
+    /**
+     * Gets the page number.
+     *
+     * @return The page number.
+     */
+    public Integer pageNumber() {
+        return pageNumber;
+    }
+
+    /**
+     * Gets the metadata.
+     *
+     * @return The metadata.
+     */
+    public Map<String, Object> metadata() {
+        return metadata;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        WebSearchInformationResult that = (WebSearchInformationResult) o;
+        return Objects.equals(totalResults, that.totalResults)
+                && Objects.equals(pageNumber, that.pageNumber)
+                && Objects.equals(metadata, that.metadata);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(totalResults, pageNumber, metadata);
+    }
+
+    @Override
+    public String toString() {
+        return "WebSearchInformationResult{" +
+                "totalResults=" + totalResults +
+                ", pageNumber=" + pageNumber +
+                ", searchInformation=" + metadata +
+                '}';
+    }
+
+    /**
+     * Creates a new WebSearchInformationResult with the specified total results.
+     *
+     * @param totalResults The total number of results.
+     * @return The new WebSearchInformationResult.
+     */
+    public static WebSearchInformationResult from(Long totalResults) {
+        return new WebSearchInformationResult(totalResults);
+    }
+
+    /**
+     * Creates a new WebSearchInformationResult with the specified total results, page number, and metadata.
+     *
+     * @param totalResults The total number of results.
+     * @param pageNumber  The page number.
+     * @param metadata     The metadata.
+     * @return The new WebSearchInformationResult.
+     */
+    public static WebSearchInformationResult from(Long totalResults, Integer pageNumber, Map<String, Object> metadata) {
+        return new WebSearchInformationResult(totalResults, pageNumber, metadata);
+    }
+
+    /**
+     * Creates a new WebSearchInformationResult with the specified total results.
+     *
+     * @param totalResults The total number of results.
+     * @return The new WebSearchInformationResult.
+     */
+    public static WebSearchInformationResult informationResult(Long totalResults) {
```

Comment:
I guess you took this approach from one of existing domain classes?

I feel like it was a bad idea to introduce so many ways to create a domain object:
- constructors
- static `ClassName.from(...)` factories
- static `className(...)` factories

I am considering deprecating the last one and not introduce new `className(...)` factories, WDYT?

Reply:
I totally agree. I no longer find a real use for it (only for unit tests perhaps, but that's it). I'll remove its.